cmd/node.go

```diff
@@ -0,0 +1,609 @@
+	Run: func(cmd *cobra.Command, args []string) {
+		stopNode()
+	},
+}
+
+var upgradeCmd = &cobra.Command{
+	Use:   "upgrade",
+	Short: "Upgrade Algod",
+	Long:  "Upgrade Algod (if installed with package manager).",
+	Run: func(cmd *cobra.Command, args []string) {
+		upgradeAlgod()
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(nodeCmd)
+	nodeCmd.AddCommand(installCmd)
+	nodeCmd.AddCommand(configureCmd)
+	nodeCmd.AddCommand(startCmd)
+	nodeCmd.AddCommand(stopCmd)
+	nodeCmd.AddCommand(upgradeCmd)
+}
+
+func installNode() {
+	fmt.Println("Checking if Algod is installed...")
+
+	// Check if Algod is installed
+	if !isAlgodInstalled() {
+		fmt.Println("Algod is not installed. Installing...")
+
+		// Install Algod based on OS
+		switch runtime.GOOS {
+		case "linux":
+			installNodeLinux()
+		case "darwin":
+			installNodeMac()
+		default:
+			panic("Unsupported OS: " + runtime.GOOS)
+		}
+	} else {
+		fmt.Println("Algod is already installed.")
+		printAlgodInfo()
+	}
+
+}
+
+func installNodeLinux() {
+	fmt.Println("Installing Algod on Linux")
+
+	// Check that we are calling with sudo
+	if !isRunningWithSudo() {
+		fmt.Println("This command must be run with super-user priviledges (sudo).")
+		os.Exit(1)
+	}
+
+	var installCmds [][]string
+	var postInstallHint string
+
+	// Based off of https://developer.algorand.org/docs/run-a-node/setup/install/#installation-with-a-package-manager
+
+	if checkCmdToolExists("apt") { // On Ubuntu and Debian we use the apt package manager
+		fmt.Println("Using apt package manager")
+		installCmds = [][]string{
+			{"apt", "update"},
+			{"apt", "install", "-y", "gnupg2", "curl", "software-properties-common"},
+			{"sh", "-c", "curl -o - https://releases.algorand.com/key.pub | tee /etc/apt/trusted.gpg.d/algorand.asc"},
+			{"sh", "-c", `add-apt-repository -y "deb [arch=amd64] https://releases.algorand.com/deb/ stable main"`},
+			{"apt", "update"},
+			{"apt", "install", "-y", "algorand-devtools"},
+		}
+	} else if checkCmdToolExists("apt-get") { // On some Debian systems we use apt-get
+		fmt.Println("Using apt-get package manager")
+		installCmds = [][]string{
+			{"apt-get", "update"},
+			{"apt-get", "install", "-y", "gnupg2", "curl", "software-properties-common"},
+			{"sh", "-c", "curl -o - https://releases.algorand.com/key.pub | tee /etc/apt/trusted.gpg.d/algorand.asc"},
+			{"sh", "-c", `add-apt-repository -y "deb [arch=amd64] https://releases.algorand.com/deb/ stable main"`},
+			{"apt-get", "update"},
+			{"apt-get", "install", "-y", "algorand-devtools"},
+		}
+	} else if checkCmdToolExists("dnf") { // On Fedora and CentOs8 there's the dnf package manager
+		fmt.Println("Using dnf package manager")
+		installCmds = [][]string{
+			{"curl", "-O", "https://releases.algorand.com/rpm/rpm_algorand.pub"},
+			{"rpmkeys", "--import", "rpm_algorand.pub"},
+			{"dnf", "install", "-y", "dnf-command(config-manager)"},
+			{"dnf", "config-manager", "--add-repo=https://releases.algorand.com/rpm/stable/algorand.repo"},
```

Comment:
Ran into the following error, Fedora 41:

```
Checking if Algod is installed...
Algod is not installed. Installing...
Installing Algod on Linux
Using dnf package manager
Running command: curl -O https://releases.algorand.com/rpm/rpm_algorand.pub
Running command: rpmkeys --import rpm_algorand.pub
Running command: dnf install -y dnf-command(config-manager)
Running command: dnf config-manager --add-repo=https://releases.algorand.com/rpm/stable/algorand.repo
Command failed: dnf config-manager --add-repo=https://releases.algorand.com/rpm/stable/algorand.repo
Output: Unknown argument "--add-repo=https://releases.algorand.com/rpm/stable/algorand.repo" for command "config-manager". Add "--help" for more information about the arguments.

Error: exit status 2

```

Reply:
Looks like it's a change upstream, the fix for 41 is:

```
dnf config-manager addrepo --from-repofile=https://releases.algorand.com/rpm/stable/algorand.repo
```

We may need to check command version numbers or the operating system distribution